### DISCUSSION TITLE
Columns should have the same order in the table editor and table update menu

### DISCUSSION THREAD

**Discussion by madebyisaacr** [supabase/supabase#12108] [Feature Requests]
In the table editor, it's possible to drag and drop column headings to reorder them. 

![image](https://user-images.githubusercontent.com/119639439/216144310-d1c04f6c-7342-45dc-aee3-d7dd148095fb.png)

In the update table menu, columns are listed in the order they were added and can't be reordered. If you add a new column, it will stay at the bottom of the list even if you move it closer to the front in the table editor.

![image](https://user-images.githubusercontent.com/119639439/216144810-10db0056-97dc-4df3-b809-677b91fb78b9.png)

If the order of the columns in the update table menu matched the order in the table editor, it would be easier and faster to find and edit table columns.

**Comment by MeyerOppelt** [comment]
Bump

**Reply by GaryAustin1** [reply]
Just an FYI the order does not change the order of returned data from Postgres.  Columns are ordered by when they were created.

**Reply by MeyerOppelt** [reply]
Yes, the feature request is that the supabase table editor will order the columns by how they are defined in postgres. The orders are loaded from a variable in browser local storage called `supabase_grid_<projectref>`. 

Grid Variable:
```
"gridColumns": [
   {
    "key": "supabase-grid-select-row",
    "name": "",
    "idx": 0,
    "width": 65,
    "maxWidth": 65,
    "resizable": false,
    "sortable": false,
    "frozen": true,
    "isLastFrozenColumn": false,
    "level": 0,
    "minWidth": 0,
    "draggable": false
   },
   {
    "key": "projectid",
    "name": "projectid",
    "idx": 1,
    "resizable": true,
    "sortable": true,
    "width": 250,
    "minWidth": 100,
    "frozen": true,
    "isLastFrozenColumn": false,
    "level": 0,
    "draggable": false
   },
   {
    "key": "created_at",
    "name": "created_at",
    "idx": 2,
    "resizable": true,
    "sortable": true,
    "width": 189,
    "minWidth": 100,
    "frozen": false,
    "isLastFrozenColumn": false,
    "level": 0,
    "draggable": false
   },
   {
    "key": "deactivated_at",
    "name": "deactivated_at",
    "idx": 3,
    "resizable": true,
    "sortable": true,
    "width": 225,
    "minWidth": 100,
    "frozen": false,
    "isLastFrozenColumn": false,
    "level": 0,
    "draggable": false
   },
   {
    "key": "project",
    "name": "project",
    "idx": 4,
    "resizable": true,
    "sortable": true,
    "width": 250,
    "minWidth": 100,
    "frozen": false,
    "isLastFrozenColumn": false,
    "level": 0,
    "draggable": false
   },
   {
    "key": "customersid",
    "name": "customersid",
    "idx": 5,
    "resizable": true,
    "sortable": true,
    "width": 250,
    "minWidth": 100,
    "frozen": false,
    "isLastFrozenColumn": false,
    "level": 0,
    "draggable": false
   },
   {
    "key": "created_by",
    "name": "created_by",
    "idx": 6,
    "resizable": true,
    "sortable": true,
    "width": 250,
    "minWidth": 100,
    "frozen": false,
    "isLastFrozenColumn": false,
    "level": 0,
    "draggable": false
   },
   {
    "key": "updated_by",
    "name": "updated_by",
    "idx": 7,
    "resizable": true,
    "sortable": true,
    "width": 250,
    "minWidth": 100,
    "frozen": false,
    "isLastFrozenColumn": false,
    "level": 0,
    "draggable": false
   },
   {
    "key": "updated_at",
    "name": "updated_at",
    "idx": 8,
    "resizable": true,
    "sortable": true,
    "width": 189,
    "minWidth": 100,
    "frozen": false,
    "isLastFrozenColumn": false,
    "level": 0,
    "draggable": false
   },
   {
    "key": "supabase-grid-add-column",
    "name": "",
    "idx": 999,
    "width": 100,
    "maxWidth": 100,
    "resizable": false,
    "sortable": false,
    "frozen": false,
    "isLastFrozenColumn": false,
    "level": 0,
    "minWidth": 0,
    "draggable": false
   }
  ]
```
![Screenshot 2024-04-06 at 4 13 47 PM](https://github.com/supabase/supabase/assets/16761291/090b67ac-4f10-4b62-b209-60d76c7bf6f3)

And my postgres definition for this table is:
```sql
create table
  schema.project (
    projectid uuid not null default gen_random_uuid (),
    created_at timestamp with time zone null,
    updated_at timestamp with time zone null,
    created_by uuid null,
    updated_by uuid null,
    deactivated_at timestamp with time zone null,
    customersid uuid not null,
    project text null,
    constraint project_pkey primary key (projectid),
    constraint project_project_key unique (project),
    constraint project_customersid_fkey foreign key (customersid) references schema.customer (customerid),
    constraint x_project_created_by_fkey foreign key (created_by) references auth.users (id),
    constraint x_project_updated_by_fkey foreign key (updated_by) references auth.users (id)
  ) tablespace pg_default;
```
![Screenshot 2024-04-06 at 4 10 48 PM](https://github.com/supabase/supabase/assets/16761291/943bb435-38ad-403a-aa35-1d83310d21bd)

Obviously you can delete this local storage key manually through browser developer tools, but I think this could be best achieved by allowing the end user to delete and reupdate `supabase_grid_<projectref>`, purging old gridcolumn data and loading it from the postgres definiton.

The only reason my gridcolumn data looks like this is because I changed the definition of the postgres table, where it was unordered in it's definition before (manually adding columns through supabase studio) and was rewritten to be in order in the definition (supabase cli migrations, dropping and rebuilding tables with the same names).
